src/app/tasks/task-details/task-details.component.ts

```diff
@@ -44,26 +48,30 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
   
   onSubmit () {
     // Get filled out form data using form group
+    let taskType : Option | undefined = this.taskTypes.find( taskType => 
+      taskType.value == this.editTaskForm.get('type')?.value
+    );
+    let taskStatus : Option | undefined = this.taskStatuses.find(taskStatus => 
+      taskStatus.value == this.editTaskForm.get('status')?.value
+    );
 
-    if(this.task?.id) {
-      this.taskService.updateTask(
-        this.task.id,
-        {
-          id: this.task.id,
-          title: this.editTaskForm.get('title')?.value,
-          description: this.editTaskForm.get('description')?.value,
-          type: this.taskTypes.find( taskType => 
-            taskType.value == this.editTaskForm.get('type')?.value
-          ) ?? null,
-          modifiedOn: new Date(),
-          createdOn: this.task.createdOn,
-          status: this.taskStatuses.find(taskStatus => 
-            taskStatus.value == this.editTaskForm.get('status')?.value
-          ) ?? null,
-        }
-      );
+    if (!taskType || !taskStatus) {
```

Comment:
same here there shouldnt be a case where a task doesnt have status ro type

Reply:
If the task type or status is undefined (because the value could not be found in the enum) then I want the component to throw an error and not update the task. Won't `throw new Error("");` achieve this?

Reply:
Yeah but there shouldn't ever be a case where it CANNOT find it, because the values are select only, and there's no possible way to input the incorrect status/type, please revise this logic, because it can be done without needing to check every time

Reply:
Okay, will look into it